index.js

```diff
@@ -5,150 +5,75 @@
- *
- * @param {∀} data Any collection-like object
  */
-function AudioBuffer (channels, data, sampleRate, options) {
-	//enforce class
-	if (!(this instanceof AudioBuffer)) return new AudioBuffer(channels, data, sampleRate, options);
-
-	//detect last argument
-	var c = arguments.length
-	while (!arguments[c] && c) c--;
-	var lastArg = arguments[c];
-
-	//figure out options
-	var ctx, isWAA, floatArray, isForcedType = false
-	if (lastArg && typeof lastArg != 'number') {
-		ctx = lastArg.context || (context && context())
-		isWAA = lastArg.isWAA != null ? lastArg.isWAA : !!(isBrowser && ctx.createBuffer)
-		floatArray = lastArg.floatArray || Float32Array
-		if (lastArg.floatArray) isForcedType = true
-	}
-	else {
-		ctx = context && context()
-		isWAA = !!ctx
-		floatArray = Float32Array
-	}
+function AudioBuffer (context, options) {
+	if (!(this instanceof AudioBuffer)) return new AudioBuffer(context, options);
 
-	//if one argument only - it is surely data or length
-	//having new AudioBuffer(2) does not make sense as 2 being number of channels
-	if (data == null || isPlainObj(data)) {
-		data = channels || 1;
-		channels = null;
-	}
-	//audioCtx.createBuffer() - complacent arguments
-	else {
-		if (typeof sampleRate == 'number') this.sampleRate = sampleRate;
-		else if (isBrowser) this.sampleRate = ctx.sampleRate;
-		if (channels != null) this.numberOfChannels = channels;
+	//if no options passed
+	if (!options) {
+		options = context
+		context = options && options.context
 	}
 
-	//if AudioBuffer(channels?, number, rate?) = create new array
-	//this is the default WAA-compatible case
-	if (typeof data === 'number') {
-		this.length = data;
-		this.data = []
-		for (var c = 0; c < this.numberOfChannels; c++) {
-			this.data[c] = new floatArray(data)
-		}
-	}
-	//if other audio buffer passed - create fast clone of it
-	//if WAA AudioBuffer - get buffer’s data (it is bounded)
-	else if (isAudioBuffer(data)) {
-		this.length = data.length;
-		if (channels == null) this.numberOfChannels = data.numberOfChannels;
-		if (sampleRate == null) this.sampleRate = data.sampleRate;
-
-		this.data = []
-
-		//copy channel's data
-		for (var c = 0, l = this.numberOfChannels; c < l; c++) {
-			this.data[c] = data.getChannelData(c).slice()
-		}
-	}
-	//TypedArray, Buffer, DataView etc, or ArrayBuffer
-	//NOTE: node 4.x+ detects Buffer as ArrayBuffer view
-	else if (ArrayBuffer.isView(data) || data instanceof ArrayBuffer || isBuffer(data)) {
-		if (isBuffer(data)) {
-			data = b2ab(data);
-		}
-		//convert non-float array to floatArray
-		if (!(data instanceof Float32Array) && !(data instanceof Float64Array)) {
-			data = new floatArray(data.buffer || data);
-		}
+	if (!options) options = {}
 
-		this.length = Math.floor(data.length / this.numberOfChannels);
-		this.data = []
-		for (var c = 0; c < this.numberOfChannels; c++) {
-			this.data[c] = data.subarray(c * this.length, (c + 1) * this.length);
-		}
+	if (context === undefined) context = getContext()
```

Comment:
What do you think about adding `context === undefined && is_browser`, so it stays undefined on Node.js? Wdyt?